src/processTargets.ts

```diff
@@ -458,13 +498,62 @@ function getLineSelectionContext(
 
   const trailingDelimiterRange =
     end.line + 1 < document.lineCount
-      ? new Range(end.line, 0, end.line + 1, 0)
+      ? new Range(end.line, end.character, end.line + 1, 0)
       : null;
 
-  // Didn't find any delimiters
-  if (leadingDelimiterRange == null && trailingDelimiterRange == null) {
+  // Outer selection contains the entire lines
+  const outerSelection = new Selection(
+    start.line,
+    0,
+    end.line,
+    selection.editor.document.lineAt(end.line).range.end.character
+  );
+
+  const isInDelimitedList =
+    leadingDelimiterRange != null || trailingDelimiterRange != null;
+
+  return {
+    isInDelimitedList,
+    containingListDelimiter: isInDelimitedList ? "\n" : undefined,
+    leadingDelimiterRange,
+    trailingDelimiterRange,
+    outerSelection,
+  };
+}
+
+function getParagraphSelectionContext(
+  selection: SelectionWithEditor,
+  selectionContext: SelectionContext
+): SelectionContext {
+  if (selectionContext.isInDelimitedList) {
     return selectionContext;
   }
+  const { document } = selection.editor;
+  const start = selection.selection.start;
+  const end = selection.selection.end;
+
+  const leadingLine =
+    start.line > 1 ? start.line - 2 : start.line > 0 ? start.line - 1 : null;
+  const trailingLine =
+    end.line + 2 < document.lineCount
+      ? end.line + 2
+      : end.line + 1 < document.lineCount
+      ? end.line + 1
+      : null;
```

Comment:
I think these should eat all empty lines until next non-empty.  This would be consistent with our space behaviour

Reply:
Agreed